### ISSUE TITLE
rc: add full set of stats to core/stats to match normal stats output

### ISSUE THREAD

**Issue by hahanonym** [rclone/rclone#5116]
#### What is your current rclone version (output from `rclone version`)?

latest master code

#### What problem are you are trying to solve?

Get sync job progress from job/status api with remote control mode

#### How do you think rclone should be changed to solve that?

In remote control mode ，user sync copy a big file，there are a long time wait and if we don't have progress， just wait no expectition。Is not a good user experience。
At present，i solve this proble with three step

1.get job target file zise
2.timer to get job/status--bytes filed
3.progress=bytes divide all file size
It's not graceful enough 。

refer to [https://forum.rclone.org/t/no-progress-return-from-job-status/22758/8](url)

**Comment by ncw** [comment]
You can get nearly all the stats but not quite. Here is a demo

Start rcd - in this case slowed down so the transfer takes some time

    rclone rcd -vv --rc-no-auth --bwlimit 1k

Start the job.

```
$ rclone rc sync/copy  _async=true _group=1 srcFs=/home/ncw/go/src/github.com/rclone/rclone dstFs=/tmp/rclone-copy
{
	"jobid": 2
}
```

You can query the stats for the job by the jobid or the group

```
$ rclone rc core/stats jobid=2
```

Which gives

```json
{
	"bytes": 192200,
	"checks": 5093,
	"deletedDirs": 0,
	"deletes": 0,
	"elapsedTime": 144.322237619,
	"errors": 0,
	"fatalError": false,
	"renames": 0,
	"retryError": false,
	"speed": 2075.375055347906,
	"transferTime": 92.609793585,
	"transferring": [
		{
			"bytes": 94208,
			"eta": 7915,
			"group": "1",
			"name": "MANUAL.html",
			"percentage": 6,
			"size": 1396481,
			"speed": 1020.1005652756488,
			"speedAvg": 164.52234094349856
		},
		{
			"bytes": 28672,
			"eta": 5686,
			"group": "1",
			"name": "MANUAL.md",
			"percentage": 2,
			"size": 1148305,
			"speed": 310.4704681521282,
			"speedAvg": 196.90610032097965
		},
		{
			"bytes": 28672,
			"eta": 9207,
			"group": "1",
			"name": "MANUAL.txt",
			"percentage": 2,
			"size": 1131642,
			"speed": 310.48486037842923,
			"speedAvg": 119.79442739419498
		},
		{
			"bytes": 6350,
			"eta": 0,
			"group": "1",
			"name": "README.md",
			"percentage": 100,
			"size": 6350,
			"speed": 236.91971820209213,
			"speedAvg": 208.11374948104577
		}
	],
	"transfers": 9
}
```

Or `rclone rc core/stats group=1` which will accumulate stats for that group.

This is **nearly** everything that you get from the normal stats.

```
$ rclone -v --stats 10s --bwlimit 1k copy ~/go/src/github.com/rclone/rclone /tmp/rclone-copy/
```

```
2021/03/12 09:48:23 INFO  : Starting bandwidth limiter at 1kBytes/s
2021/03/12 09:48:33 INFO  : 
Transferred:   	   24.135k / 967.953 MBytes, 0%, 2.414 kBytes/s, ETA 4d18h2m31s
Checks:                13 / 13, 100%
Transferred:            0 / 5084, 0%
Elapsed time:        10.0s
Transferring:
 *                                    RELEASE.md:100% /2.487k, 282/s, 0s
 *                           RELEASE.md.~master~:100% /2.497k, 284/s, 0s
 *                                    TODO-v1.51:100% /19.143k, 2.126k/s, 0s
 *                                       VERSION:100% /8, 0/s, 0s
```

I'll **bold** the things that I think are missing

> Transferred:   	   24.135k / **967.953 MBytes**, **0%**, 2.414 kBytes/s, **ETA 4d18h2m31s**
> Checks:                13 / **13**, **100%**
> Transferred:            0 / **5084**, **0%**
> Elapsed time:        10.0s
> Transferring:
>                                      RELEASE.md:100% /2.487k, 282/s, 0s
>                             RELEASE.md.master:100% /2.497k, 284/s, 0s
>                                      TODO-v1.51:100% /19.143k, 2.126k/s, 0s
>                                         VERSION:100% /8, 0/s, 0s
> 

So I think some of the the whole transfer stats are missing, namely

- total size (and percentage)
- eta
- total checks (and percentage)
- total transferred (and percentage)

Are missing.

Would you agree?

**Comment by hahanonym** [comment]
Yes，job/status add progress is not a good way。
I also agree  add fileds to stats。
Another question，if there are total percent in stats，Do I still need to call job/status to know whether finish or success？

**Comment by ncw** [comment]
> Yes，job/status add progress is not a good way。
> I also agree add fileds to stats。

I updated the title to to change the output of core/stats

> Another question，if there are total percent in stats，Do I still need to call job/status to know whether finish or success？

Yes you should still call job/status

**Comment by ncw** [comment]
Give this a go

[v1.55.0-beta.5271.f9ef9c0ab.fix-5116-rc-stats](https://beta.rclone.org/branch/fix-5116-rc-stats/v1.55.0-beta.5271.f9ef9c0ab.fix-5116-rc-stats/) on branch [fix-5116-rc-stats](https://github.com/rclone/rclone/tree/fix-5116-rc-stats) (uploaded in 15-30 mins)

I've added these stats

    - totalChecks - total number of checks needed in sync
    - totalTransfers - total number of transfers needed in sync
    - totalBytes - total number of bytes to transfer in sync
    - eta - eta of sync in seconds

**Comment by hahanonym** [comment]
It worked。thank you。

**Comment by ivandeex** [comment]
This has been resolved, or do I miss something?

**Comment by ivandeex** [comment]
I think it also implements #2280

**Comment by hahanonym** [comment]
> I think it also implements #2280

I think #2280 miss some filed。I mind sync job status。
I have reused already exist field bytes。and need more info to know the job progress。
This issue solve my problem。

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.55
